backport: v20.0.2 backports and release (#5750)
dashpay/dash: doc/release-notes.md

# Notable changes

## Masternode fix

> A problem has been fixed in old quorum data cleanup mechanism that was slowing down Masternode during DKGs causing them to get PoSe scored.

Reviewer: ```suggestion
A problem has been fixed in the old quorum data cleanup mechanism. It was slowing down masternodes during DKG sessions and causing them to get PoSe scored.
```

## Testnet Crash

> A crash has been fixed which was reported when updating from v19.x to v20.0.0 after v20 activation without re-indexing.

Reviewer: ```suggestion
A fix has been implemented for the reported crash that could occur when upgrading from v19.x to v20.0.0 after v20 activation without re-indexing.
```

## Guix Build System Enhancements

> The Guix build system has been enhanced with caching and increased timeouts.

Reviewer 2: That's for Github CI only.

Author: So we shouldn't mention that?

Reviewer 2: Could maybe make it `Other changes` section and say smth like `Implemented improvements in Github CI and build system for macOS. Fixed compilation issues on FreeBSD.`

These release are considered obsolete. Old release notes can be found here:

> - [v20.0.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-20.0.1.md) released November/15/2023

Reviewer 2: ```suggestion
- [v20.0.1](https://github.com/dashpay/dash/blob/master/doc/release-notes/dash/release-notes-20.0.1.md) released November/18/2023
```
